ehitchhiking-ui/src/app/shared/components/rate-passengers-modal/rate-passengers-modal.component.ts

```diff
@@ -2,60 +2,113 @@ import {Component, Inject, OnInit} from '@angular/core';
-  }
-
-  ngOnInit(): void {
-    // here should be just GET 'info' about trip
-    if (this.data.dataKey === UserState.Driver){
-      this.users = [
-        {id: 0, name: 'Ivan', rating: 0, isAddedToBlackList: false},
-        {id: 1, name: 'Egor', rating: 0, isAddedToBlackList: false},
-        {id: 2, name: 'Semen', rating: 0, isAddedToBlackList: false},
-        {id: 3, name: 'Fedor', rating: 0, isAddedToBlackList: false},
-      ];
-    }
-    else{
-      this.users = [{id: 0, name: 'Ivan', rating: 0, isAddedToBlackList: false}];
-    }
-
-
-  }
-
-  rateUser(clickObj: StarClickMeta): void {
-		const user = this.users.find((i: User) => i.id === clickObj.itemId);
-		if (!!user) {
-			user.rating = clickObj.rating;
+export class RatePassengersModalComponent implements OnInit {
+	users: RateUser[] = [];
+	UserState = UserState;
+	idTripDriver = 12;
+	idTripPassenger = 4;
+	loading = true;
+	loaderSize: LoaderSize = LoaderSize.Large;
+	constructor(
+		public dialogRef: MatDialogRef<RatePassengersModalComponent>,
+		@Inject(MAT_DIALOG_DATA) public data: any,
+		private apiRatePassengersService: RatePassengersApiService
+	) {}
+
+	ngOnInit(): void {
+		if (this.data.dataKey === UserState.Passenger) {
+			this.loadDriversList();
+		} else {
+			this.loadPassengersList();
+		}
+	}
+
+	loadPassengersList() {
+		this.loading = true;
+		this.apiRatePassengersService.getTripPassengers(this.idTripPassenger).then((data) => {
+			this.users = data;
+			this.loading = false;
+		});
+	}
+
+	loadDriversList(): void {
+		this.loading = true;
+		this.apiRatePassengersService.getTripDriver(this.idTripDriver).then((data) => {
+			this.users = data;
+			this.loading = false;
+		});
+	}
+
+	rateUser(clickObj: StarClickMeta): void {
+		if (this.data.dataKey === UserState.Passenger) {
+			this.users.forEach((i: RateUser) => {
+				i.id === clickObj.itemId ? (i.rate = clickObj.rating) : '';
+				return i;
+			});
+		} else {
+			this.users.forEach((i: RateUser) => {
+				i.id === clickObj.itemId ? (i.rate = clickObj.rating) : '';
+				return i;
+			});
 		}
 	}
 
 	exitTrip(): void {
+		const drivers = [];
+		const pass = [];
+		const blockedPass = [];
+		const blockedDrivers = [];
+		this.users.forEach((user) => {
+			drivers.push({
+				idDriver: user.id,
+				rate: user.rate,
+			});
+			pass.push({
+				idPass: user.id,
+				rate: user.rate,
+			});
+			blockedDrivers.push({
+				id: user.id,
+				isBlocked: user.isBlocked,
+			});
+			blockedPass.push({
+				id: user.id,
+				isBlocked: user.isBlocked,
+			});
+		});
+		if (this.data.dataKey === UserState.Passenger) {
+			this.apiRatePassengersService.addRateDriver(drivers);
+			this.apiRatePassengersService.addBlacklistDriver(this.idTripDriver, blockedPass);
+		} else {
+			this.apiRatePassengersService.addRatePassenger(pass);
+			this.apiRatePassengersService.addBlacklistPass(this.idTripPassenger, blockedDrivers);
+		}
 		this.dialogRef.close();
```

Comment:
you should not close modal until rests finished successfully

Reply:
done